lib/widgets/mixins/twake_context_menu_mixin.dart

```diff
@@ -1,27 +1,123 @@
-import 'package:fluffychat/pages/chat/events/message/message_content_with_timestamp_builder.dart';
+import 'package:fluffychat/widgets/context_menu/context_menu_action.dart';
 import 'package:fluffychat/widgets/context_menu/twake_context_menu.dart';
+import 'package:fluffychat/widgets/mixins/twake_context_menu_style.dart';
+import 'package:fluffychat/widgets/twake_app.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
 
 /// Show a [TwakeContextMenu] on the given [BuildContext]. For other parameters, see [TwakeContextMenu].
 mixin TwakeContextMenuMixin {
-  void showTwakeContextMenu({
+  Future<int?> showTwakeContextMenu({
+    required List<ContextMenuAction> listActions,
     required Offset offset,
     required BuildContext context,
-    required ContextMenuBuilder builder,
     double? verticalPadding,
     VoidCallback? onClose,
   }) async {
-    await showDialog(
+    int? result;
+    await showDialog<int>(
       context: context,
       barrierColor: Colors.transparent,
-      barrierDismissible: true,
-      builder: (context) => TwakeContextMenu(
+      barrierDismissible: false,
+      builder: (dialogContext) => TwakeContextMenu(
+        dialogContext: dialogContext,
+        listActions: listActions,
         position: offset,
-        builder: builder,
         verticalPadding: verticalPadding,
       ),
     ).then((value) {
+      result = value;
       onClose?.call();
     });
+    return result;
+  }
+
+  Widget contextMenuItem(
```

Comment:
Why need it? I think duplocated with `ContextMenuActionItemWidget`

Reply:
updated